lib/dangermattic/plugins/common/common_release_checks.rb

```diff
@@ -0,0 +1,100 @@
+    #
+    # @param fail_on_error [Boolean] If true, a failure message will be displayed instead of a warning.
+    #
+    # @example Check if any modified file is under the 'app/' directory and emit a warning on release branches:
+    #   check_file_changed(file_comparison: ->(file_path) { file_path.include?('app/') },
+    #                      message: 'Some files in the "app/" directory have been modified. Please review the changes.',
+    #                      on_release: true
+    #   )
+    #
+    # @example Check if a specific file has been modified and emit a failure on non-release branches:
+    #   check_file_changed(file_comparison: ->(file_path) { file_path == 'path/to/file/DoNotChange.java' },
+    #                      message: 'The "DoNotChange.java" file has been modified. This change is not allowed on non-release branches.',
+    #                      on_release: false,
+    #                      fail_on_error: true)
+    #
+    def check_file_changed(file_comparison:, message:, on_release:, fail_on_error: false)
+      has_modified_file = all_modified_files.any?(&file_comparison)
+
+      should_be_changed = on_release ? release_branch? : !release_branch?
+      return unless should_be_changed && has_modified_file
+
+      if fail_on_error
+        failure(message)
+      else
+        warn(message)
+      end
+    end
+
+    # Check if the release notes and store strings files are correctly updated after a modification to the release notes.
+    #
+    # @param release_notes_file [String] The name of the release notes file that should be checked for modifications.
+    #   Example: 'metadata/release_notes.txt'
+    #
+    # @param store_strings_file [String] The name of the store strings file that should be checked for modifications.
+    #   Example: 'metadata/PlayStoreStrings.po'
+    #
+    # @param fail_on_error [Boolean] If true, a failure message will be emitted instead of a warning.
+    #
+    # @example Check if the release notes file 'release_notes.txt' is modified, and the 'PlayStoreStrings.po' file is not modified:
+    #   check_release_notes_and_store_strings(release_notes_file: 'release_notes.txt', store_strings_file: 'PlayStoreStrings.po', fail_on_error: false)
+    #
+    def check_release_notes_and_store_strings(release_notes_file:, store_strings_file:, fail_on_error: false)
+      has_modified_release_notes = danger.git.modified_files.any? { |f| f.end_with?(release_notes_file) }
+      has_modified_app_store_strings = danger.git.modified_files.any? { |f| f.end_with?(store_strings_file) }
+
+      return unless has_modified_release_notes && !has_modified_app_store_strings
+
+      message = "The #{File.basename(store_strings_file)} file must be updated any time changes are made to the release notes."
+
+      if fail_on_error
+        failure(message)
+      else
+        warn(message)
+      end
+    end
```

Comment:
I'm wondering if this old Peril Rule still has value nowadays given that:

1️⃣  it has always triggered a false positive in the 2-step process we follow during WP code freeze, where:
1. Usually the `metadata/release_notes.txt` gets changed during code freeze to be filled with the bare/raw/unedited release notes extracted from the top section of `RELEASE-NOTES.txt`, but at that point we don't want to update the `metadata/*StoreStrings.po` file with those because that would send unedited/raw release notes to GlotPress
2. And only when our freelance writer has massaged and wordsmithed a nice, editorialized release note text from those bare entries, do we update the `metadata/release_notes.txt` file with the nice copy, and only then run the lane that will update the `.po` file with that nice copy to be sent to GlotPress for translation

This means that historically, that Peril rule has always been triggered during step 1, and we've learned to ignore it as a false positive for the particular case of this step.

2️⃣  Also, with @oguzkocer working on migrating code freeze and other release operations to CI, I'm wondering if the future new process that will come from that will have an impact on those steps (1) and (2) above; in particular, if step (1) is done fully in CI via a `code_freeze/…` intermediate branch in the future, maybe there won't be a PR for step (1) anymore then? (in which case, if we will only have a PR in step (2), modifying both `release_notes.txt` and `*.po`, then maybe the rule will still have value now that the false positive from step 1 would be no more?

Interested to get the insights and thoughts from others @Automattic/apps-infrastructure , and in particular from folks who are doing Release Driving for any of the apps that rely on those `*.po` file + GlotPress (i.e. to my knowledge, at least WP/JP/WC… maybe others?)

Reply:
As a side note, @mokagio mentioned ( paaHJt-5bL-p2#comment-7740 ) they use the `Releases` label to avoid running these rules (thus avoiding the false positive warning) as a release manager.

Reply:
Pinging again @Automattic/apps-infrastructure / @oguzkocer -- any thoughts to share on this?

Reply:
> 2️⃣ Also, with @oguzkocer working on migrating code freeze and other release operations to CI, I'm wondering if the future new process that will come from that will have an impact on those steps (1) and (2) above; in particular, if step (1) is done fully in CI via a code_freeze/… intermediate branch in the future, maybe there won't be a PR for step (1) anymore then? (in which case, if we will only have a PR in step (2), modifying both release_notes.txt and *.po, then maybe the rule will still have value now that the false positive from step 1 would be no more?

Regardless of whether the code freeze happens in CI or in our local environment, there has to be a PR to merge the created release branch to `trunk`. So, the release management in CI changes shouldn't impact this.

---

In my experience, whenever I saw this warning it was always a false positive. That might be because release managers know when the `.po` files should be updated, but when we start rotating product developers to do releases, they would not have this knowledge. Our release scenarios are very detailed, so this is unlikely to be a problem, but it may still be useful to share this info in PRs just in case.

So, if we can work on the messaging and make this an info rather than an error or a warning, I think that'd probably provide the most value without getting in the way.

> message = "The #{File.basename(store_strings_file)} file must be updated any time changes are made to the release notes."

Maybe instead of this, it should say `.po` file `should` be updated `if the editorialized release notes are changed`. Or we can be more verbose about it when to update the `.po` file and when not to do it. 🤷

Reply:
Thanks for the suggestion, @oguzkocer (sorry for the delay)! I've slightly changed the message and the type of report (just a regular `message` now): 29d8949627a2f60827c6497b85bd0662c228ed2e